examples/edb_cfg_examples/setup_edb_for_signal_integrity_analysis.py

```diff
@@ -0,0 +1,186 @@
+# 2:accuracy.
+# - **freq_sweep**. List of frequency sweeps.
+#   - **name**. Name of the sweep.
+#   - **type**. Type of the sweep. Supported types are 'interpolation', 'discrete', 'broadband'.
+#   - **frequencies**. Frequency distribution.
+#     - **distribution**. Supported distributions are 'linear_count', 'linear_scale', 'log_scale'.
+#     - **start**. Start frequency. Example, 1e6, "1MHz".
+#     - **stop**. Stop frequency. Example, 1e9, "1GHz".
+#     - **increment**.
+
+cfg["setups"] = [
+    {
+        "name": "siwave_syz",
+        "type": "siwave_ac",
+        "pi_slider_position": 1,
+        "freq_sweep": [
+            {
+                "name": "Sweep1",
+                "type": "interpolation",
+                "frequencies": [{"distribution": "log_scale", "start": 1e6, "stop": 1e9, "increment": 20}],
+            }
+        ],
+    }
+]
+
+# ### Define Cutout
+# Keywords
+# - **signal_list**. List of nets to be kept after cutout.
+# - **reference_list**. List of nets as reference planes.
+# - **extent_type**. Supported extend types are 'Conforming', 'ConvexHull', 'Bounding'.
+# For optional input arguments, refer to method pyedb.Edb.cutout()
+
+cfg["operations"] = {
+    "cutout": {
+        "signal_list": ["1V0"],
+        "reference_list": ["GND"],
+        "extent_type": "ConvexHull",
+    }
+}
+
+# ## Save the configuration as a JSON file
+# The configuration file can be saved in JSON format and applied to layout data using the EDB.
+
+file_json = os.path.join(temp_folder.name, "edb_configuration.json")
+with open(file_json, "w") as f:
+    json.dump(cfg, f, indent=4, ensure_ascii=False)
+
+# ## Load configuration into EDB
+
+# Load layout.
+
+edbapp = Edb(aedb, edbversion=AEDT_VERSION)
+
+# Load configuration file
+
+edbapp.configuration.load(config_file=file_json)
+
+# Apply configuration to EDB.
+
+edbapp.configuration.run()
+
+# Save and close EDB.
+
+edbapp.save()
+edbapp.close()
+
+# The configured EDB file is saved in a temp folder.
+
+print(temp_folder.name)
+
+# ## Analyze in HFSS 3D Layout
+
+# ### Load edb into HFSS 3D Layout.
+
+h3d = Hfss3dLayout(aedb, version=AEDT_VERSION, non_graphical=NG_MODE, new_desktop=True)
+
+# ### Analyze
+
+h3d.analyze()
+
+# ### Plot impedance
+
+solutions = h3d.post.get_solution_data(expressions="Z(port1,port1)")
+solutions.plot()
+
+# ## Shut Down Electronics Desktop
+
+h3d.close_desktop()
+
+# All project files are saved in the folder ``temp_file.dir``. If you've run this example as a Jupyter notbook you
+# can retrieve those project files. The following cell removes all temporary files, including the project folder.
+
+# ## Cleanup
+#
+# All project files are saved in the folder ``temp_file.dir``. If you've run this example as a Jupyter notbook you
+# can retrieve those project files. The following cell removes all temporary files, including the project folder.
+
+time.sleep(3)
```

Comment:
Not sure this is required anymore. There was an issue with the desktop close call but I think it got fixed recently.

Reply:
Then, I will remove it.